src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java

```diff
@@ -0,0 +1,288 @@
+            APPIUM_FOLDER + File.separator + BIN_FOLDER + File.separator + APPIUM_JS;
+
+    public static final String APPIUM_NODE_PROPERTY = "appium.node.path";
+    public static final String DEFAULT_LOCAL_IP_ADDRESS = "0.0.0.0";
+
+    private static final int DEFAULT_APPIUM_PORT = 4723;
+
+    private final static String COMMAND_WHICH_EXTRACTS_DEFAULT_PATH_TO_APPIUM = "npm -g ls --depth=0";
+    private final static String COMMAND_WHICH_EXTRACTS_DEFAULT_PATH_TO_APPIUM_WIN = "npm.cmd -g ls --depth=0";
+
+    private static final int REQUIRED_MAJOR_NODE_JS = 0;
+    private static final int REQUIRED_MINOR_NODE_JS = 0;
+
+    final Map<String, String> serverArguments = new HashMap<>();
+    private File appiumJS;
+    private String ipAddress = DEFAULT_LOCAL_IP_ADDRESS;
+
+    private long timeToWaitForAppiumNodeIsStarted = 60;
+    private TimeUnit timeUnitToWaitForAppiumNodeIsStarted = TimeUnit.SECONDS;
+
+
+    private static String returnCommandThatSearchesForDefaultNode(){
+        if (Platform.getCurrent().is(Platform.WINDOWS))
+            return COMMAND_WHICH_EXTRACTS_DEFAULT_PATH_TO_APPIUM_WIN;
+        return COMMAND_WHICH_EXTRACTS_DEFAULT_PATH_TO_APPIUM;
+    }
+
+    private static String getProcessOutput(InputStream stream ) throws IOException {
+        BufferedReader reader = new BufferedReader(
+                new InputStreamReader(stream));
+        String result = reader.readLine();
+        reader.close();
+        return result;
+    }
+
+    private static void validateNodeJSVersion(){
+        Runtime rt = Runtime.getRuntime();
+        String result = null;
+        try {
+            Process p = rt.exec("node -v");
+            p.waitFor();
+            result = getProcessOutput(p.getInputStream());
+        } catch (Exception e) {
+            throw new InvalidNodeJSInstance("Node.js is not installed", e);
+        }
+        String versionNum =  result.replace("v","");
+        String[] tokens = versionNum.split("\\.");
+        if (Integer.parseInt(tokens[0]) < REQUIRED_MAJOR_NODE_JS ||
+                Integer.parseInt(tokens[1]) < REQUIRED_MINOR_NODE_JS)
+            throw new InvalidNodeJSInstance("Current node.js version " + versionNum + "is lower than " +
+                    "required (0.10 or greater)");
+    }
+
+    private static File findNodeInCurrentFileSystem(){
+        Runtime rt = Runtime.getRuntime();
+        String instancePath;
+        try {
+            Process p = rt.exec(returnCommandThatSearchesForDefaultNode());
+            p.waitFor();
+            instancePath = getProcessOutput(p.getInputStream());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        File result;
+        if (StringUtils.isBlank(instancePath) || !(result = new File(instancePath + File.separator + NODE_MODULES_FOLDER +
+                APPIUM_NODE_MASK)).exists())
+            throw new InvalidServerInstanceException( "There is no installed nodes! Please install " +
```

Comment:
I think this error message should be referenced as a String instead of duplicated.